src/gurobi_optimods/metromap.py

```diff
@@ -0,0 +1,1242 @@
+            graph, model, edge_direction, x, y, mindist, num_nodes + mindist
+        )
+
+        model.setObjective(obj[0])
+        model.update()
+
+        if include_planarity:
+            # helper variables for ensuring planarity
+            startEdges = []
+            gamma = {}
+            for u1, v1 in graph.edges:
+                startEdges.append((u1, v1))
+                for u2, v2 in graph.edges:
+                    if (u2, v2) in startEdges:
+                        continue
+                    if u1 == u2 or u1 == v2 or v1 == u2 or v1 == v2:
+                        continue
+                    for i in range(8):
+                        gamma[u1, v1, u2, v2, i] = model.addVar(
+                            vtype="b", name=f"planarity[{u1},{v1},{u2},{v2},{i}]"
+                        )
+                    objGamma = model.addVar(
+                        lb=0, ub=1, name=f"objGamma[{u1},{v1},{u2},{v2}]"
+                    )
+                    model.addConstr(
+                        gp.quicksum(gamma[u1, v1, u2, v2, i] for i in range(8))
+                        + objGamma
+                        == 1
+                    )
+                    obj[0] += 1000 * (objGamma)
+
+            # necessary data for planarity callback
+            model._mindist = mindist
+            model._x = x
+            model._y = y
+            model._gamma = gamma
+            model._bigM = num_nodes + num_nodes * mindist
+            model._graph = graph
+            model.setParam("LazyConstraints", 1)
+            # optimize with callback for planarity constraints,
+            # faster than adding all of them from the beginning
+            model.optimize(planarity_callback)
+        else:
+            model.optimize()
+
+        if model.solCount > 0:
+            for v in graph.nodes:
+                graph.add_node(v, pos_oct=(x[v].X, y[v].X))
+
+            for u, v in graph.edges:
+                for i in range(8):
+                    if edge_direction[u, v, i].X > 0.5:
+                        edge_directions_out[(u, v)] = i
+                    if edge_direction[v, u, i].X > 0.5:
+                        edge_directions_out[(v, u)] = i
+
+    return graph, edge_directions_out
+
+
+# Helper function
+def counter_clockwise_sort(p, q):
+    # Sort the positions given in q counter clockwise around p, start from east
+
+    # Enumerate q to remember the original order
+    q = list(enumerate(q))
+
+    # Define a nested function for sorting
+    def sort_by_angle(item):
+        index, point = item
+        dx, dy = point[0] - p[0], point[1] - p[1]
+        angle = math.atan2(dy, dx)
+        # Adjust the angle to start from East direction
+        if angle < 0:
+            angle += 2 * math.pi
+        return angle
+
+    # Sort the points counter-clockwise
+    q.sort(key=sort_by_angle)
+
+    # Return the sorted points along with their original indices
+    return q
+
+
+# Assume geographical node data is given.
```

Comment:
Can we move these function comments to the under the function definition? Additionally, if these are all private functions you are not expecting to expose you can just prepend them with `_` e.g.  `_create_orig...`. For ref see also: [src/gurobi_optimods/bipartite_matching.py#L96](https://github.com/Gurobi/gurobi-optimods/blob/main/src/gurobi_optimods/bipartite_matching.py#L96)

Reply:
I tried to synchronize this. Please check again.